src/Connector/GceServiceAccountAuthentication.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Connector;
+
+use Apigee\Edge\ClientInterface;
+use Apigee\Edge\Exception\HybridOauth2AuthenticationException;
+use Apigee\Edge\HttpClient\Plugin\Authentication\AbstractOauth;
+use Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface;
+use GuzzleHttp\Exception\ConnectException;
+use Http\Message\Authentication\Header;
+
+/**
+ * Decorator for Hybrid authentication plugin.
+ */
+class GceServiceAccountAuthentication extends AbstractOauth {
```

Comment:
Hm, my gut says that this connector should be added to the [PHP API client ](https://github.com/apigee/apigee-client-php)as the Drupal module is just one of the possible PHP consumers.

Reply:
Good catch I added moved over a lot of the code to the PHP API Client.

Reply:
Just for the reference :) https://github.com/apigee/apigee-client-php/pull/107